pandas/core/indexes/datetimelike.py

```diff
@@ -191,10 +194,7 @@ def sort_values(self, return_indexer=False, ascending=True):
             sorted_index = self.take(_as)
             return sorted_index, _as
         else:
-            # NB: using asi8 instead of _ndarray_values matters in numpy 1.18
-            #  because the treatment of NaT has been changed to put NaT last
-            #  instead of first.
-            sorted_values = np.sort(self.asi8)
+            sorted_values = np.sort(self._ndarray_values)
```

Comment:
this means i8n for PeriodIndex (so NaT to the front), m8 for TimedeltaIndex (so NaT to the front until numpy updates td64 to match dt64) and M8 for DatetimeIndex, (so NaT to the end).  Is that what you're going for here?

Reply:
NumPy has fixed the timedelta sorting, at least for 1.19+

```python
In [22]: arr = np.array([np.timedelta64(2, 'ns'), np.timedelta64('NaT', 'ns'), np.timedelta64(1, 'ns')])

In [23]: arr.sort()

In [24]: arr
Out[24]: array([    1,     2, 'NaT'], dtype='timedelta64[ns]')
```